From 6287b07506ec463f851222fba0d29634fc1c9de5 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:25 -0400
Subject: [fs] nfsd: Use simple_read_from_buffer for coping data to userspace

Message-id: <1403560314-32525-84-git-send-email-bfields@redhat.com>
Patchwork-id: 84509
O-Subject: [Patch RHEL7 083/172] NFSD: Use simple_read_from_buffer for coping data to userspace
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: f3e41ec5ef0f5d2e10b6bfd3a13dc29f6d260d79
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/fault_inject.c b/fs/nfsd/fault_inject.c
index d620e7f..2ed05c3 100644
--- a/fs/nfsd/fault_inject.c
+++ b/fs/nfsd/fault_inject.c
@@ -97,25 +97,14 @@ static ssize_t fault_inject_read(struct file *file, char __user *buf,
 {
  static u64 val;
  char read_buf[25];
- size_t size, ret;
+ size_t size;
  loff_t pos = *ppos;
 
  if (!pos)
   nfsd_inject_get(file_inode(file)->i_private, &val);
  size = scnprintf(read_buf, sizeof(read_buf), "%llu\n", val);
 
- if (pos < 0)
-  return -EINVAL;
- if (pos >= size || !len)
-  return 0;
- if (len > size - pos)
-  len = size - pos;
- ret = copy_to_user(buf, read_buf + pos, len);
- if (ret == len)
-  return -EFAULT;
- len -= ret;
- *ppos = pos + len;
- return len;
+ return simple_read_from_buffer(buf, len, ppos, read_buf, size);
 }
 
 static ssize_t fault_inject_write(struct file *file, const char __user *buf,
-- 
1.7.1